website/content/docs/k8s/installation/vault/data-integration/webhook-certs.mdx

```diff
@@ -0,0 +1,261 @@
+path controller/issue/controller-role {
+  capabilities = ["create", "update"]
+}
+EOF
+```
+
+```shell-session
+$ vault policy write connect-inject-policy - <<EOF
+path connect-inject/issue/connect-inject-role {
+  capabilities = ["create", "update"]
+}
+EOF
+```
+#### Create Vault Policies for the CA URL
+
+-> **Note:** The PKI secret paths referenced by the Vault Policies below will be your `global.secretsBackend.vault.controller.caCert.secretName` and `global.secretsBackend.vault.connectInject.caCert.secretName` Helm values respectively.
+
+Next, we will create a policy that allows `["read"]` access to the [CA URL](https://www.vaultproject.io/api/secret/pki#read-certificate),
+this is required for the Consul components to communicate with the Consul servers in order to fetch their auto-encryption certificates.
+
+```shell-session
+$ vault policy write controller-ca-policy - <<EOF
+path controller/cert/ca {
+  capabilities = ["read"]
+}
+EOF
+```
+
+```shell-session
+$ vault policy write connect-inject-ca-policy - <<EOF
+path connect-inject/cert/ca {
+  capabilities = ["read"]
+}
+EOF
+```
+
+## Setup per Consul datacenter
+### Create a Vault PKI role that establishes the domains that it is allowed to issue certificates for
+
+Create a Vault role for the controller PKI engine will set the default certificate issuance parameters:
+
+```shell-session
+$ vault write controller/roles/controller-role \
+    allowed_domains="<Allowed-domains-string>" \
+    allow_subdomains=true \
+    allow_bare_domains=true \
+    allow_localhost=true \
+    generate_lease=true \
+    max_ttl="720h"
+```
+
+Create a Vault role for the connect inject PKI engine will set the default certificate issuance parameters:
+
+```shell-session
+$ vault write connect-inject/roles/connect-inject-role \
+    allowed_domains="<Allowed-domains-string>" \
+    allow_subdomains=true \
+    allow_bare_domains=true \
+    allow_localhost=true \
+    generate_lease=true \
+    max_ttl="720h"
+```
+
+To generate the `<Allowed-domains-string>` for each use the following script as a template:
+
+```shell-session
+#!/bin/sh
+
+# NAME is set to either the value from `global.name` from your Consul K8s value file, or your $HELM_RELEASE_NAME-consul
+export NAME=consulk8s
+# NAMESPACE is where the Consul on Kubernetes is installed
+export NAMESPACE=consul
+# DATACENTER is the value of `global.datacenter` from your Helm values config file
+export DATACENTER=dc1
+
+echo allowed_domains_controller=\"${NAME}-controller-webhook,${NAME}-controller-webhook.${NAMESPACE},${NAME}-controller-webhook.${NAMESPACE}.svc,${NAME}-controller-webhook.${NAMESPACE}.svc.cluster.local\""
+
+echo allowed_domains_connect_inject=\"${NAME}-connect-injector,${NAME}-connect-injector.${NAMESPACE},${NAME}-connect-injector.${NAMESPACE}.svc,${NAME}-connect-injector.${NAMESPACE}.svc.cluster.local\""
+```
+
+### Create a Vault auth roles that link the policy to each Consul on Kubernetes service account that requires access
```

Comment:
```suggestion
### Link the Vault policy to Consul workloads
Create a Vault auth roles that link the policy to each Consul workload on Kubernetes service account that requires access.
```
Is it workloads? It wasn't clear what "each Consul" meant. Cluster?

Reply:
yes.  thank you.  updated in several places.